engine/src/main/java/com/redhat/cloud/notifications/processors/drawer/DrawerProcessor.java

```diff
@@ -0,0 +1,138 @@
+
+    @Inject
+    TemplateService templateService;
+
+    @Inject
+    BaseTransformer baseTransformer;
+
+    @Inject
+    ObjectMapper objectMapper;
+
+    @Inject
+    EndpointRepository endpointRepository;
+
+    @Inject
+    DrawerNotificationRepository drawerNotificationRepository;
+
+    @Inject
+    EventRepository eventRepository;
+
+    @Inject
+    FeatureFlipper featureFlipper;
+
+    @Override
+    public void process(Event event, List<Endpoint> endpoints) {
+        if (!featureFlipper.isDrawerEnabled()) {
+            return;
+        }
+        if (endpoints != null && !endpoints.isEmpty()) {
+            Set<User> userList = getRecipientList(event, endpoints, EmailSubscriptionType.DRAWER);
+            if (null != userList && !userList.isEmpty()) {
+                process(event, userList);
+            }
+        }
+
+    }
+
+    private void process(Event event, Set<User> userList) {
+        UUID historyId = UUID.randomUUID();
+        Endpoint endpoint = endpointRepository.getOrCreateDefaultSystemSubscription(event.getAccountId(), event.getOrgId(), EndpointType.DRAWER);
+        Log.infof("Processing drawer notification [orgId=%s, eventId=%s, historyId=%s]",
+            endpoint.getOrgId(), event.getId(), historyId);
+
+        long startTime = System.currentTimeMillis();
+
+        NotificationHistory history = null;
+        try {
+            String userNameListAsStr = userList.stream().map(usr -> usr.getUsername()).collect(Collectors.joining(","));
+            List<DrawerNotification> drawerNotifications = drawerNotificationRepository.create(event, userNameListAsStr);
+
+            // build event thought qute template
+            String renderedData = buildNotificationMessage(event);
+            // store it on event table
+            event.setRenderedDrawerNotification(renderedData);
+            eventRepository.updateDrawerNotification(event);
+            endpoint = endpointRepository.getOrCreateDefaultSystemSubscription(event.getAccountId(), event.getOrgId(), EndpointType.DRAWER);
+
+            // TODO push created drawerNotifications through kafta: RHCLOUD-25999
+
+            history = getHistoryStub(endpoint, event, 0L, historyId);
+            history.setStatus(NotificationStatus.SUCCESS);
+        } catch (Exception e) {
+            history = getHistoryStub(endpoint, event, 0L, historyId);
+            history.setStatus(NotificationStatus.FAILED_INTERNAL);
+            history.setDetails(Map.of("failure", e.getMessage()));
+            Log.infof(e, "Processing drawer notification failed [eventId=%s, historyId=%s]", event.getId(), historyId);
+        } finally {
+            long invocationTime = System.currentTimeMillis() - startTime;
+            history.setInvocationTime(invocationTime);
+            persistNotificationHistory(history);
+        }
+    }
+
+    public String buildNotificationMessage(Event event) {
+        JsonObject data = baseTransformer.toJsonObject(event);
```

Comment:
We'll probably need to add the `Environment#url` to `data`. We can take care of that later.